app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -213,12 +219,14 @@ const SurveyAnimals: React.FC = () => {
           .test('checkDeploymentRangeEnd', '', async (value, context) => {
```

Comment:
move this schema to device.ts.
Might be reused in the future.

Reply:
I don't think it's possible to easily insert this into device.ts, as it is not a react component. The call to deploymentOverlapTest uses the telemetryApi hook, and hooks cannot be used outside of components.

Reply:
I did look at moving these schema into device.ts, and yeah I don't think it would be very straightforward given that it relies on those API hooks